opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -167,21 +176,27 @@ public ExprValue construct(String jsonString) {
    * @param value value object
    * @return ExprValue
    */
-  public ExprValue construct(String field, Object value) {
-    return parse(new ObjectContent(value), field, type(field));
+  public ExprValue construct(String field, Object value, boolean supportArrays) {
+    return parse(new ObjectContent(value), field, type(field), supportArrays);
   }
 
-  private ExprValue parse(Content content, String field, Optional<ExprType> fieldType) {
+  private ExprValue parse(
+      Content content,
+      String field,
+      Optional<ExprType> fieldType,
+      boolean supportArrays
+  ) {
     if (content.isNull() || !fieldType.isPresent()) {
       return ExprNullValue.of();
     }
 
     ExprType type = fieldType.get();
-    if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.Object))
-          || type == STRUCT) {
-      return parseStruct(content, field);
-    } else if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.Nested))) {
-      return parseArray(content, field);
+    if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.Nested))
+        || content.isArray()) {
+      return parseArray(content, field, type, supportArrays);
+    } else if (type.equals(OpenSearchDataType.of(OpenSearchDataType.MappingType.Object))
+        || type == STRUCT) {
+      return parseStruct(content, field, supportArrays);
     } else {
```

Comment:
nit: else not required.

Reply:
Without this else nothing will be parsed. Both `parseStruct` and `parseArray` will end their recurse in this `else`.